proto/v1alpha1/rode.proto

```diff
@@ -95,6 +100,10 @@ service Rode {
   }
 }
 
+message GenericResource {
+  string name = 1;
```

Comment:
Throwing this idea out there because I'm trying to wrap my head around how the UI will consume this new endpoint.  Would it be possible (or a good idea) to return the latest version of the resource with this call?  I don't think it needs to be stored in the database, but maybe pulled whenever the endpoint gets called.

I'm thinking of the scenario to search for a resource.
1. Type in a search term, call this endpoint to get the list of possible resources.
1. When you select a resource to view, the UI will call the `listResources` endpoint to get all possible versions and pick the latest created version.
1. Call the `listOccurrences` endpoint to get the occurrences, then show the UI.

Returning the latest version would eliminate the second step. I could maybe see this endpoint returning a list of all versions too, then the UI only needs to string together the resource name and version to get any data it needs.

Reply:
I think it would be possible:
1. multi-search with the resource name as the prefix in the query
2. sort by created or updated on the occurrence
3. limit the size to 1

 It would definitely be useful to the UI, but it seems a little strange to have it in the data model since it's always computed and not a trait of the resource itself but derived from all of the occurrences.

Reply:
Yeah, I agree it is a little strange but it would certainly cut down on some work that the UI needs.  I'm fine either way, I can work with the given model, just wanted to throw it out there!

Reply:
I like this idea, definitely don't need to include it in this PR. Essentially upon every creation of an occurrence, we are either creating or updating a version of a resource.

Reply:
Whether it's in Rode or the UI, taking the newest might not work as expected, since we don't have a distinct concept of the resources, only what we can aggregate from occurrences. There could be a recent prod deployment occurrence that has an older version of the resource but is itself the most recent occurrence. 

We could could limit the search to just build occurrences, although that might not exist for every resource (thinking about 3rd party resources that might be put into a registry and scanned but not rebuilt from source).

Reply:
Good point, that definitely makes things tricky. This seems like a good topic to discuss with the larger group. I like the idea of basing it off build occurrences. If a resource doesn't have build occurrences, we could use the most recent occurrence, regardless of the occurrence type. Maybe that could be some signal to the front end that this resource is treated differently than resources that are entirely managed by the team and could be represented in a different way.